numpy/linalg/linalg.py

```diff
@@ -1749,8 +1749,9 @@ def pinv(a, rcond=1e-15 ):
     """
     a, wrap = _makearray(a)
     rcond = asarray(rcond)
+    m, n = a.shape[-2:]
```

Comment:
these variables are only used in a branch, will that change?
if not move them into the branch

Reply:
I think I'd just keep the original.

Reply:
@juliantaylor: good catch, will move

@mhvk: I prefer it this way because it makes it clearer that:
* it matches the documented (n, m) return value
* the shape has been reversed. I find it easy to misread `a.shape[-1], a.shape[-2]` as non-reversed.

If you don't agree with that, I'll change it back.